seisgo/noise.py

```diff
@@ -384,55 +379,218 @@ def do_correlation(sfile,win_len,step,maxlag,cc_method='xcorr',acorr_only=False,
+            try:
+                loc1=get_locator(ds,sta_list[ii])
+                loc.append(loc1)
+            except Exception as e:
+                print('abort! no stationxml for %s in file %s'%(sta_list[ii],sfile))
+                continue
+        tt4=time.time()
+        ttt1+=tt4-tt3
+        #print('###################################################### Read inventory takes {:.5f}'.format(tt4-tt3))
+        
+        # Loop through all source stations
+        for iiS in range(len(sta_list)):
+            # get source station and inventory
+            issta=sta_list[iiS]
+            print('source: {}'.format(issta))
+            
+            # check if the station has assembled raw data
+            if issta in data_cache:
+                source=data_cache[issta]
+            else:
+                # get days information: works better than just list the tags
+                sall_tags = ds.waveforms[issta].get_waveform_tags()
+                sloc=loc[iiS][0]
+                sinv1=loc[iiS][1]
+                #fftdata=[]
+                #get source raw data
+                tt1=time.time()
+                source=assemble_raw(sfile,ds,sall_tags,sloc,issta,sinv1,v,correct_orientation=correct_orientation,pad_thre=pad_thre,do_rotation=do_rotation)
+                tt2=time.time()
+                ttt2+=tt2-tt1
+                #print('##################################################### Assemble source raw data takes {:.5f}'.format(tt2-tt1))
+                if source is None:
+                    print('CONTINUE! Something wrong when assembling raw data.')
+                    #sys.exit()
+                    continue
+                
+                data_cache[issta]=source
+            
+            # if rotation is True, loop through all receiver stations, do rotation and then do fft and cross-correlation
+            if do_rotation:
+                iend=len(sta_list)
+                istart=iiS
+                src=issta
+                # loop through all receiver stations
+                for iiR in range(istart,iend):
+                    rcv=sta_list[iiR]
+                    irsta=rcv
+                    print('receiver: {}'.format(irsta))
+                    
+                    if irsta in data_cache:
+                        receiver=data_cache[irsta]
+                    else:
+                        rall_tags = ds.waveforms[irsta].get_waveform_tags()
+                        rloc=loc[iiR][0]
+                        rinv1=loc[iiR][1]
+                        # get receiver raw data
+                        tt11=time.time()
+                        receiver=assemble_raw(sfile,ds,rall_tags,rloc,irsta,rinv1,v,correct_orientation=correct_orientation,pad_thre=pad_thre,do_rotation=do_rotation)
+                        tt12=time.time()
+                        ttt6+=tt12-tt11
+                        if receiver is None:
+                            print('CONTINUE! Something wrong when assembling raw data.')
```

Comment:
be a bit more specific in error message. e.g., no data assembled.

Reply:
Same here, when a None value is returned by assemble_raw(), the corresponding error message will be printed.
The one you mentioned here is only for identifying which station has a problem.